client/network/src/protocol/notifications/upgrade/notifications.rs

```diff
@@ -247,11 +250,15 @@ where TSubstream: AsyncRead + AsyncWrite + Unpin,
 					},
 
 				st @ NotificationsInSubstreamHandshake::NotSent |
-				st @ NotificationsInSubstreamHandshake::Normal { .. } |
-				st @ NotificationsInSubstreamHandshake::BothSidesClosed => {
+				st @ NotificationsInSubstreamHandshake::Normal { .. } => {
 					*this.handshake = st;
 					return Poll::Pending;
```

Comment:
Not familiar with the code enough. The `Pending` in this case is fine?

Reply:
Indeed that's not fine.